lib/matplotlib/tests/test_sphinxext.py

```diff
@@ -122,6 +122,56 @@ def test_plot_html_show_source_link(tmpdir):
     assert "index-1.py" not in [p.name for p in html_dir2.iterdir()]
 
 
+def test_show_source_link_true(tmpdir):
+    source_dir = Path(tmpdir) / 'src'
+    source_dir.mkdir()
+    parent = Path(__file__).parent
+    shutil.copyfile(parent / 'tinypages/conf.py', source_dir / 'conf.py')
+    shutil.copytree(parent / 'tinypages/_static', source_dir / '_static')
+    doctree_dir = source_dir / 'doctrees'
+    (source_dir / 'index.rst').write_text("""
+.. plot::
+    :show-source-link: true
+
+    plt.plot(range(2))
+""")
+    # Make sure source scripts are created by default
+    html_dir1 = source_dir / '_build' / 'html1'
+    build_sphinx_html(source_dir, doctree_dir, html_dir1)
+    assert "index-1.py" in [p.name for p in html_dir1.iterdir()]
+    # Make sure source scripts are still created when
+    # plot_html_show_source_link` is False
+    html_dir2 = source_dir / '_build' / 'html2'
+    build_sphinx_html(source_dir, doctree_dir, html_dir2,
+                      extra_args=['-D', 'plot_html_show_source_link=0'])
+    assert "index-1.py" in [p.name for p in html_dir1.iterdir()]
+
+
+def test_show_source_link_false(tmpdir):
+    source_dir = Path(tmpdir) / 'src'
+    source_dir.mkdir()
+    parent = Path(__file__).parent
+    shutil.copyfile(parent / 'tinypages/conf.py', source_dir / 'conf.py')
+    shutil.copytree(parent / 'tinypages/_static', source_dir / '_static')
+    doctree_dir = source_dir / 'doctrees'
+    (source_dir / 'index.rst').write_text("""
+.. plot::
+    :show-source-link: false
+
+    plt.plot(range(2))
+""")
+    # Make sure source scripts are NOT created
+    html_dir1 = source_dir / '_build' / 'html1'
+    build_sphinx_html(source_dir, doctree_dir, html_dir1)
+    assert "index-1.py" not in [p.name for p in html_dir1.iterdir()]
+    # Make sure source scripts are still NOT created when
+    # plot_html_show_source_link` is False
+    html_dir2 = source_dir / '_build' / 'html2'
+    build_sphinx_html(source_dir, doctree_dir, html_dir2,
+                      extra_args=['-D', 'plot_html_show_source_link=0'])
```

Comment:
Shouldn't this be the reverse
```suggestion
    # Make sure source scripts are still NOT created even if
    # plot_html_show_source_link` is True
    html_dir2 = source_dir / '_build' / 'html2'
    build_sphinx_html(source_dir, doctree_dir, html_dir2,
                      extra_args=['-D', 'plot_html_show_source_link=1'])
```
The core logic to test here is

> if `:show-source-link:'` is set, `plot_html_show_source_link` should be ignored

So in theory, one could test both values of `plot_html_show_source_link`. However, even with an incorrect implementation, most likely no source link would be shown if both parameters are set to false. So the important test is for different values.

Reply:
I refactored the test. Please take another look.